src/pages/en/guides/content-collections.mdx

```diff
@@ -187,290 +217,211 @@ defineCollection({
 
-```astro "{ headings }"
+Call [`getCollection()`](/en/reference/api-reference/#getcollection) inside of `getStaticPaths()` to query your content. Then, create your new URL paths using the `slug` property of each content entry.
+
+```astro "{ slug: entry.slug }"
 ---
+// src/pages/posts/[slug].astro
 import { getCollection } from 'astro:content';
-const blogPosts = await getCollection('blog');
+// 1. Generate a new path for every collection entry
+export async function getStaticPaths() {
+  const blogEntries = await getCollection('blog');
+  return blogEntries.map(entry => ({
+    params: { slug: entry.slug }, props: { entry },
+  }));
+}
+// 2. When its time to render, you can get the entry directly from the prop
+const { entry } = Astro.props;
+const { Content } = await entry.render();
 ---
-
-{blogPosts.map(async (post) => {
-  const { headings } = await post.render();
-  const h1 = headings.find(h => h.depth === 1);
-  return <p>{h1}</p>
-})}
+<h1>{entry.data.title}</h1>
+<Content />
 ```
 
-### Access modified frontmatter from `render()`
+This will generate a new page for every entry in the `blog` collection. For example, an entry at `src/content/blog/hello-world.md` will have a slug of `hello-world`, and therefore its final URL will be `/posts/hello-world/`.
 
-Astro allows you to [modify frontmatter using remark or rehype plugins](/en/guides/markdown-content/#modifying-frontmatter-programmatically). You can access this modified frontmatter using the `remarkPluginFrontmatter` property from `render()`:
+### Building for server output (SSR)
 
-```astro "{ remarkPluginFrontmatter }"
+If you are building a dynamic website (using Astro's SSR support), you are not expected to generate any paths ahead-of-time during the build. Instead, your page should examine the request (using `Astro.request` or `Astro.params`) to find the `slug` on-demand, and then fetch it using [`getEntryBySlug()`](/en/reference/api-reference/#getentrybyslug). 
+
+
+```astro "{ slug: Astro.params }"
 ---
-import { getCollection } from 'astro:content';
-const blogPosts = await getCollection('blog');
+// src/pages/posts/[slug].astro
+// NOTE: This example will only work in 
+import { getEntryBySlug } from 'astro:content';
+// 1. Get the slug from the request, on-demand
+const { slug } = Astro.params;
+// 2. Query the entry directly using its slug
+const entry = await getEntryBySlug('blog', slug);
+// 3. (Optional) render the entry to HTML in the template
+const { Content } = await entry.render();
 ---
-
-{blogPosts.map(async (post) => {
-  const { remarkPluginFrontmatter } = await post.render();
-  return <p>{post.data.title} — {remarkPluginFrontmatter.readingTime}</p>
-})}
+<h1>{entry.data.title}</h1>
+<Content />
 ```
 
-Assuming `readingTime` was injected ([see our reading time example](/en/guides/markdown-content/#example-calculate-reading-time)), it will be available on the `remarkPluginFrontmatter` object.
+## Modifying Frontmatter with Remark
```

Comment:
This is an appropriate use of the caution! Something we don't actually want anyone to do!  :smile: